File: linked-lists/single/deletion/src/main/java/com/yauritux/single/model/Node.java
package com.yauritux.single.model;

/**
 * @author yauritux
 */
public final class Node {

    private String name;
    private Node next;

    public Node() {}

    public Node(final String name ){
        this.name = name;
    }

    public String getName() {
	return name;
    }

    public void setName(final String name) {
	this.name = name;
    }

    public Node getNext() {
	return next;
    }

    public void setNext(Node next) {
	this.next = next;
    }

    @Override
    public String toString() {
	return name; 
    }
}

File: typescript/linked-list/single/Node.ts
export default class Node {
  private value: string;
  private next: Node | null;

  constructor(value: string) {
    this.value = value;
    this.next = null;
  }

  append(value: string): Node {
    this.next = new Node(value);
    return this.next;
  }

  print(): void {
    let current: Node = this;
    let output = current.value;
    while (current.next) {
      output += " -> " + current.next.value;
      current = current.next;
    }
    console.log(output);
  }
}
